transitions/core.py

```diff
@@ -376,12 +380,37 @@ def __init__(self, model=None, states=None, initial=None, transitions=None,
         if ordered_transitions:
             self.add_ordered_transitions()
 
-        for model in self.models:
-            if hasattr(model, 'trigger'):
-                logger.warning("%sModel already contains an attribute 'trigger'. Skip method binding ",
-                               self.id)
-            else:
-                model.trigger = partial(get_trigger, model)
+        if model:
+            self.add_model(model)
+        elif add_self:
+            self.add_model(self)
+
+    def add_model(self, model):
```

Comment:
Actually, it looks like some of the other public methods also don't have docstrings, so you can ignore this and I'll do a pass later.